src/pages/index.js

```diff
@@ -4,73 +4,197 @@ import Section from "../components/Section.js";
+      "Content-Type": "application/json",
+    },
+  });
 
-const profileEditForm = new PopupWithForm("#profile-edit-popup", (data) => {
-  userInformation.setUserInfo({
-    title: data.header,
-    description: data.description,
+  const userInformation = new UserInfo({
+    name: ".profile__header",
+    description: ".profile__description",
+    avatar: ".profile__avatar",
   });
-});
-profileEditForm.setEventListeners();
-
-constants.profileEditButton.addEventListener("click", () => {
-  profileEditValidator.resetValidation();
-  const userData = userInformation.getUserInfo();
-  constants.profileTitleInput.value = userData.title;
-  constants.profileDescriptionInput.value = userData.description.trim();
-  profileEditForm.open();
-});
 
-const addCardForm = new PopupWithForm("#add-card-popup", (data) => {
-  cardSection.addItem(createCard({ name: data.title, link: data.URL }));
-  addCardValidator.disableButton();
-});
-addCardForm.setEventListeners();
+  let cardSection;
+  api
+    .getAppInfo()
+    .then(([userInfo, initialCards]) => {
+      userInformation.setUserInfo(userInfo);
+      userInformation.setUserAvatar(userInfo);
+      if (Array.isArray(initialCards)) {
+        cardSection = new Section({ renderer: renderCard }, ".cards__list");
+        cardSection.renderItems(initialCards);
+      } else {
+        console.error("Initial cards data is not an array:", initialCards);
+      }
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 
-constants.addCardButton.addEventListener("click", () => {
-  addCardForm.open();
-});
+  function renderCard(cardData) {
+    const card = new Card(
+      cardData,
+      "#card-template",
+      handleImageClick,
+      handleCardDeleteClick,
+      handleLikeClick
+    ).getView();
+    cardSection.addItem(card);
+  }
 
-const cardSection = new Section(
-  {
-    renderer: (item) => {
-      const cardElement = createCard(item);
-      cardSection.addItem(cardElement);
-    },
-  },
-  constants.cardListEl
-);
+  const cardPreview = new PopupWithImage("#image-popup");
+  const confirmDeletePopup = new PopupWithConfirm("#delete-card-popup");
+  cardPreview.setEventListeners();
+
+  const profileEditValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#profile-form")
+  );
+  const addCardValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#add-card-form")
+  );
+  const avatarValidator = new FormValidator(
+    constants.config,
+    document.querySelector("#avatar-form")
+  );
+
+  profileEditValidator.enableValidation();
+  addCardValidator.enableValidation();
+  avatarValidator.enableValidation();
+
+  const avatarImagePopup = new PopupWithForm(
+    "#edit-avatar-popup",
+    handleAvatarFormSubmit
+  );
+
+  avatarImagePopup.setEventListeners();
+  document.querySelector(".profile__avatar").addEventListener("click", () => {
+    avatarImagePopup.open();
```

Comment:
Please, make method `toggleButtonState` **public** by removing the low dash and call it on openings to work around the enabling the button on `finally`


```js

    avatarValidator.toggleButtonState();

```